### DISCUSSION TITLE
Playing sound: publishing on cmd_audio topic does not work as described in the docs

### DISCUSSION THREAD

**Discussion by ilarioazzollini** [iRobotEducation/create3_docs#119] [Questions]
Following the documentation, I tried to play sound through the speakers as described here:

https://iroboteducation.github.io/create3_docs/api/ui/#playing-sound-through-the-speakers

It looks like the example related to publishing a corresponding message on the cmd_audio topic does not work (nothing happens). While, the example using the audio_note_sequence action server works just fine.

**Comment by justinIRBT** [comment]
Doing pub —once from the command line can be flaky in terms of the message getting through.  Try removing the once to have it publish multiple times and it should work.  I would expect a publication with a reliable qos in an application to be more robust about getting through to the subscription

**Reply by ilarioazzollini** [reply]
If I remove the `--once` it is still not able to produce any sound with the first publish message. It starts making sound after some time (publishing # 3).

Then, I tried changing the reliability of the publisher as follows

```bash
ros2 topic pub --once --qos-reliability reliable /cmd_audio irobot_create_msgs/msg/AudioNoteVector "{append: false, notes: [{frequency: 100, max_runtime: {sec: 1,nanosec: 0}}, {frequency: 50, max_runtime: {sec: 1,nanosec: 0}}]}"
```
but it still doesn't work. The reason should be that the /cmd_audio topic subscription has a "best_effort" reliability (checked using `ros2 topic info /cmd_audio --verbose`).

**Reply by alsora** [reply]
Hi @ilarioazzollini, we just released updated firmwares both for Galactic ([G.4.4](https://iroboteducation.github.io/create3_docs/releases/g_4_4/)) and Humble ([H.1.1](https://iroboteducation.github.io/create3_docs/releases/h_1_1/)) which include a change to the `cmd_audio` subscription that is located on the robot.

Now the subscription uses a reliable QoS.
This means that you should not have the problem anymore when sending single messages there from the command line.

Note that this is a breaking change: a best-effort publisher won't be able to communicate anymore with this subscription.
The publisher that is used under the hood when you do `ros2 topic pub` is of reliable type, so this doesn't apply there.
